src/Components/Components/src/ComponentResolver.cs

```diff
@@ -13,61 +14,40 @@ namespace Microsoft.AspNetCore.Components
     /// </summary>
     internal static class ComponentResolver
     {
-        /// <summary>
-        /// Lists all the types
-        /// </summary>
-        /// <param name="appAssembly"></param>
-        /// <returns></returns>
+        private static readonly string ComponentAssemblyName = typeof(IComponent).Assembly.FullName;
+        private static readonly ConcurrentDictionary<Assembly, Type[]> Cache =
+            new ConcurrentDictionary<Assembly, Type[]>();
+
         public static IEnumerable<Type> ResolveComponents(Assembly appAssembly)
```

Comment:
Is there any benefit to returning a faster type here? Would have to look at how this is used.

Reply:
I was looking at the one caller for this and it seems like we could go ahead and cache the route entries rather than the component types which are effectively an intermediate result. Is there a reason we wouldn't want to cache those instead?

Change https://github.com/aspnet/AspNetCore/blob/master/src/Components/Components/src/Routing/Router.cs#L68 to `RouteEntryCache.GetRouteEntries`?